### PR TITLE
Optimizer: CharClassClassrangesMerge: Only merge \w chars and char codes into class ranges

### LINKED ISSUE #156: Wrong classranges optimization
Something is wrong with classranges optimizations, example here:

    $ bin/regexp-tree -e '/[^a-zA-Z0-9_!#$%&*@＠]/' -o
   
    Optimized: /[^\d!#$%&*@-Z_a-z＠]

With `charClassClassrangesMerge` optimization disabled I have better result:

    Expected:  /[^\w!#$%&*@＠]/

### PR DESCRIPTION
**Pull request by Golmote**
This fixes #156.

EDIT: Somehow I did not get notifications for the discussion that happened in #156 between yesterday evening and this morning. I still think this PR might be relevant though, since we already had a special case to only merge word chars and char codes in a class range [in some cases](https://github.com/DmitrySoshnikov/regexp-tree/pull/157/files#diff-48e74382c584f8474d8aadcdb8af4beeR162).

### CODE REVIEW

File: src/optimizer/transforms/__tests__/char-class-classranges-merge-transform-test.js [hunk 1]
```diff
@@ -100,6 +100,13 @@ describe('char-class-classranges-merge', () => {
     expect(re.toString()).toBe('/[\\ud83d\\ude80-\\ud83d\\ude88]/u');
   });
 
+  it('does not merge non \\w chars into class ranges to keep readability', () => {
```
Comment (silentroach) [inline src/optimizer/transforms/__tests__/char-class-classranges-merge-transform-test.js]: but why? optimization isn't about readability

Comment (Golmote) [reply src/optimizer/transforms/__tests__/char-class-classranges-merge-transform-test.js]: At the moment, the optimizer tries to find the right balance between optimization, strictness and readability. Once we add options, we'll be able to have more atomic settings.

Comment (DmitrySoshnikov) [reply src/optimizer/transforms/__tests__/char-class-classranges-merge-transform-test.js]: Yeah, I think it's a good trade-off, thanks guys!

**Comment by davisjam** [comment]
+1

**Pull request merged**